client/src/components/history/history-chat.tsx

```diff
@@ -0,0 +1,168 @@
+import { makeStyles } from "@material-ui/core/styles";
+import { visibilitySwitch } from "store/actions";
+
+import { FormGroup, FormControlLabel, Switch } from "@material-ui/core";
+
+import { ChatData, ChatMsg, State, Config } from "types";
+import "styles/history-chat.css";
+import ChatItem, { ChatItemData } from "./history-item";
+// import { DialpadRounded } from "@material-ui/icons";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "auto",
+  },
+  list: {
+    marginTop: 1,
+    padding: 10,
+    maxHeight: "20vh",
+  },
+  avatar: {
+    width: theme.spacing(4),
+    height: theme.spacing(4),
+    backgroundColor: "#88929e",
+  },
+  GOOD: {
+    backgroundColor: "#00bf00",
+  },
+  BAD: {
+    backgroundColor: "#E63535",
+  },
+  icon: {
+    position: "absolute",
+    right: -40,
+  },
+  feedbackPopup: {
+    borderRadius: "30px",
+  },
+  chat_container: {
+    backgroundColor: "#fff",
+  },
+  introMsg: {
+    marginLeft: "0rem !important",
+  },
+  visibilityBtn: {
+    marginLeft: "5px",
+  },
+}));
+
+interface ScrollingQuestionsParams {
+  height: number;
+  questionHistory: string[];
+}
+
+const MENTOR_COLORS = ["#cfe0f9", "#FFEBE7", "#69af86", "#ffeb94"];
+
+export function HistoryChat(args: ScrollingQuestionsParams): JSX.Element {
+  const { height } = args;
+  const styles = useStyles();
+
+  const colorByMentorId = useSelector<State, Record<string, string>>((s) => {
+    const mentorIds = Object.getOwnPropertyNames(s.mentorsById);
+    mentorIds.sort();
+    return mentorIds.reduce<Record<string, string>>((acc, cur, i) => {
+      acc[cur] = MENTOR_COLORS[i % MENTOR_COLORS.length];
+      return acc;
+    }, {});
+  });
+  const namesByMentorId = useSelector<State, Record<string, string>>((s) => {
+    const mentorIds = Object.getOwnPropertyNames(s.mentorsById);
+    mentorIds.sort();
+    return mentorIds.reduce<Record<string, string>>((acc, cur) => {
+      acc[cur] = s.mentorsById[cur].mentor.name;
+      return acc;
+    }, {});
+  });
+
+  const chatData = useSelector<State, ChatData>((s) => s.chat);
+  const configData = useSelector<State, Config>((s) => s.config);
+
+  const dispatch = useDispatch();
+  const [checked, toggleChecked] = useState<boolean>(true);
+
+  const totalMentors = configData.mentorsDefault.length;
+
+  useEffect(() => {
+    animateScroll.scrollToBottom({
+      containerId: "chat-thread",
+    });
+  }, [chatData.messages]);
+
+  const toggleAnswers = (
+    <div>
+      <FormGroup className="togglePos">
+        <FormControlLabel
+          control={
+            <Switch
+              size="medium"
+              checked={checked}
+              onChange={() => toggleChecked((prev) => !prev)}
+              data-cy="visibility-switch"
+            />
+          }
+          label="hide/show answers"
+        />
+      </FormGroup>
+    </div>
+  );
+
+  useEffect(() => {
+    dispatch(visibilitySwitch(chatData.messages.length, checked));
```

Comment:
what does this do? Does it toggle visibility of answers on/off for all messages or just for one?

Reply:
it toggles the visibility of answers on/off for all message